pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -715,13 +716,21 @@ private int getOrCreateDocId(Map<String, Object> dictIdMap) {
     }
 
     // All dimension columns should be dictionary encoded.
+    // All dimensions must be single value
     for (String dimension : schema.getDimensionNames()) {
       if (noDictionaryColumns.contains(dimension)) {
         _logger
             .warn("Metrics aggregation cannot be turned ON in presence of no-dictionary dimensions, eg: {}", dimension);
         _aggregateMetrics = false;
         break;
       }
+      // https://github.com/apache/incubator-pinot/issues/3867
+      if (!schema.getDimensionSpec(dimension).isSingleValueField()) {
```

Comment:
This limitation should apply to multi-valued metrics as well, right?

Reply:
i don't think so. The line where it's breaking right now is MutableSegmentImpl::672
```
for (String column : _schema.getDimensionNames()) {
      dictIds[i++] = (Integer) dictIdMap.get(column);
    }
```
It's checked only for dimensions and time. Unless I'm missing some other place?

Reply:
Well the aggregation code also assumes the metric is single valued, so it will fail, if I am not wrong. For example, the line below is casting the value to Integer, and would fail if the value was an int[].
indexReaderWriter.setInt(docId, (Integer) value + indexReaderWriter.getInt(docId));